src/cpp/src/continuous_batching/pipeline_base.cpp

```diff
@@ -227,9 +231,13 @@ ContinuousBatchingPipeline::IContinuousBatchingPipeline::generate(
     if (m_is_chat_conversation) {
         if (encoded_results[0].m_status == ov::genai::GenerationStatus::CANCEL) {
             m_history.pop_back();
+            for (size_t i = 0; i < encode_images_num; i++) {
+                m_history_images.pop_back();
+            }
         }
         else {
             m_history.push_back({{"role", "assistant"}, {"content", results[0].texts[0]}});
+            m_image_id += rgbs_vector[0].size();
```

Comment:
```suggestion
            m_image_id += encoded_images.size();
```
So `encode_images_num` isn't needed

Reply:
Yes, replaced with `encoded_images.size()`.